src/main/webapp/app/guided-tour/guided-tour.service.ts

```diff
@@ -0,0 +1,367 @@
+import { ErrorHandler, Injectable } from '@angular/core';
+import { HttpClient, HttpResponse } from '@angular/common/http';
+import { cloneDeep } from 'lodash';
+import { JhiAlertService } from 'ng-jhipster';
+import { fromEvent, Observable, of, Subject } from 'rxjs';
+import { debounceTime } from 'rxjs/internal/operators';
+
+import { SERVER_API_URL } from 'app/app.constants';
+import { courseOverviewTour } from 'app/guided-tour/tours/course-overview-tour';
+import { GuidedTourSettings } from 'app/guided-tour/guided-tour-settings.model';
+import { ContentType, GuidedTour, Orientation, OrientationConfiguration, TourStep } from './guided-tour.constants';
+import { AccountService } from 'app/core';
+
+export type EntityResponseType = HttpResponse<GuidedTourSettings>;
+
+@Injectable()
+export class GuidedTourService {
+    public resourceUrl = SERVER_API_URL + 'api/guided-tour-settings';
+
+    public guidedTourCurrentStepStream: Observable<TourStep>;
```

Comment:
It's never good to give direct access to a class attribute, e.g. making it possible to override it.
I think it would be better to use a getter instead.